parser/mpFuncCommon.cpp

```diff
@@ -957,4 +985,57 @@ MUP_NAMESPACE_START
     return new FunWeekYear(*this);
   }
 
+  //------------------------------------------------------------------------------
+  //                                                                             |
+  //            Function return the week day of a date                           |
+  //            Usage: weekday("2022-04-20")                                     |
+  //            Optional locale: weekday("2022-04-20", "en")                     |
+  //                                                                             |
+  //------------------------------------------------------------------------------
+
+  FunWeekDay::FunWeekDay()
+    :ICallback(cmFUNC, _T("weekday"), -1)
+  {}
+
+  void FunWeekDay::Eval(ptr_val_type &ret, const ptr_val_type *a_pArg, int a_iArgc)
+  {
+    if (a_iArgc < 1) {
+      throw ParserError(ErrorContext(ecTOO_FEW_PARAMS, GetExprPos(), GetIdent()));
+    } else if (a_iArgc > 2) {
+      throw ParserError(ErrorContext(ecTOO_MANY_PARAMS, GetExprPos(), GetIdent()));
+    }

```

Comment:
Hmmmm! We can write variable arity functions! This will make our case strategy simpler!